spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java

```diff
@@ -202,53 +208,54 @@ public long executeTask(String taskName, Map<String, String> taskDeploymentPrope
 			}
 		}
 
+		// Build execution info and stash deploy props there
 		TaskExecutionInformation taskExecutionInformation = taskExecutionInfoService
 				.findTaskExecutionInformation(taskName, taskDeploymentProperties, composedTaskRunnerName);
-
 		if (taskExecutionInformation.isComposed()) {
 			handleAccessToken(commandLineArgs, taskExecutionInformation);
 		}
 
+		// Build app deploy request and stash deploy props there
 		TaskExecution taskExecution = taskExecutionRepositoryService.createTaskExecution(taskName);
-
 		AppDeploymentRequest appDeploymentRequest = this.taskAppDeploymentRequestCreator.
```

Comment:
Can we create the `AppDeploymentRequest` later in the method instead of updating it again?

Reply:
This flow is so complex with so many side effects even in a original code that for now you kinda need to. Eventually these methods need a complete rewrite.